sdk/test-utils/recorder/README.md

```diff
@@ -1,9 +1,450 @@
+`TEST_MODE`. If the environment variable `TEST_MODE` is empty, `record` (and most
+of the functions provided by test-utils-recorder) won't be doing anything. You'll need
+to set this environment variable to `record` to start recording, and then to
+`playback` to play the recordings back at your code.
+
+#### package.json scripts
+
+You can write scripts in your `package.json` to
+make it easier to switch from record mode to playback mode, on a meaningful context, as follows:
+
+    "integration-test:node": "mocha myNodeTests.js",
+    "unit-test:node": "TEST_MODE=playback npm run integration-test:node",
+    "test:node:record": "TEST_MODE=record npm run integration-test:node"
+
+
+Once your tests run, new files will be created in the `recordings/*` folder.
+These files will have names that are relative to the tests that you have.
+There might be cases in which the recordins get outdated with the test files, so you might also want to
+add a way to clear the recordings on your `package.json`, like the following one:
+
+    "clear-recordings": "rm -fr recordings",
+
+#### Karma configuration
+
+To make sure you're able to do record and playback on your browser tests, make
+sure to configure Karma properly.
+
+The recordings are separated between NodeJS recordings and browser recordings,
+so to use them on Karma, You'll at least need to add the recordings to your
+`files` array in your `karma.conf.js`, as follows:
+
+    files: [
+      // ... you might have other things here. Keep them.
+      "recordings/browsers/**/*.json"
+    ],
+
+Same goes on the `preprocessors` array:
+
+    preprocessors: {
+      // ... you might have other things here. Keep them.
+      "recordings/browsers/**/*.json": ["json"]
+    },
+
+You can also tell Karma to hide console.logs in during the playbacks in the browsers by adding
+the following configuration property:
+
+    browserConsoleLogOptions: {
+      terminal: process.env.TEST_MODE !== "record"
+    },
+
+#### Environment variables
+
+Since we make use of the `TEST_MODE` environment variables, we recommend you to
+take control of how you deal with environment variables for your tests. One way
+of doing this is with the [dotenv](https://www.npmjs.com/package/dotenv)
+project, which is what we mostly use in our packages. dotenv comes with its own
+issues, since it might act unexpectedly in relation with already assigned
+environment variables. dotenv doesn't replace any environment variable that is
+set by another source, so keep that in mind.
+
+For how to hide sensitive information related to environment variables, check
+out our example called [Securing sensitive data](#securing-sensitive-data).
+
+## Examples
+
+### How to record
+
+To record your tests, make sure to call to the `record` function exported from
+the test-utils-recorder, then call it before the http request you want to make
+happens. In the following example, we'll be starting to record before
+authenticating our KeyVault client:
```

Comment:
test-utils-recorder -> @azure/test-utils-recorder

[Maybe replace all the occurrences in this document at once.]


`then call it before the http request you want to make
happens.` ? -> `then call it before the http request you want to make`